ndcube/utils/cube.py

```diff
@@ -119,3 +123,88 @@ def convert_extra_coords_dict_to_input_format(extra_coords, missing_axis):
                 raise KeyError("extra coords dict can have keys 'wcs axis' or 'axis'.  Not both.")
             result.append((name, axis, extra_coords[name]["value"]))
         return result
+
+
+def _get_pixel_quantities_for_dependent_axes(dependent_axes, cube_dimensions):
```

Comment:
This function scares me a little!! What is it trying to do?

Reply:
We should replace this with ```np.meshgrid(..., indexing='ij')```